mlflow/genai/judges/tools/search_trace_regex.py

```diff
@@ -0,0 +1,144 @@
+                    "Search through the entire trace using a regular expression pattern. "
+                    "Performs case-insensitive matching across all trace fields including spans, "
+                    "metadata, tags, requests, and responses. Returns all matches with surrounding "
+                    "context. Useful for finding specific patterns, values, or text anywhere in "
+                    "the trace."
+                ),
+                parameters=ToolParamsSchema(
+                    type="object",
+                    properties={
+                        "pattern": {
+                            "type": "string",
+                            "description": (
+                                "Regular expression pattern to search for. The search is "
+                                "case-insensitive. Examples: 'error.*timeout', 'user_id:\\s*\\d+', "
+                                "'function_name\\(.*\\)'"
+                            ),
+                        },
+                        "max_matches": {
+                            "type": "integer",
+                            "description": "Maximum number of matches to return (default: 50)",
+                            "default": 50,
+                        },
+                    },
+                    required=["pattern"],
+                ),
+            ),
+            type="function",
+        )
+
+    def invoke(self, trace: Trace, pattern: str, max_matches: int = 50) -> SearchTraceRegexResult:
+        """
+        Search through the trace using a regex pattern.
+
+        Args:
+            trace: The MLflow trace object to search through
+            pattern: Regular expression pattern to search for
+            max_matches: Maximum number of matches to return
+
+        Returns:
+            SearchTraceRegexResult containing the search results
+        """
+        try:
+            regex = re.compile(pattern, re.IGNORECASE)
+        except re.error as e:
+            return SearchTraceRegexResult(
+                pattern=pattern,
+                total_matches=0,
+                matches=[],
+                error=f"Invalid regex pattern: {e}",
+            )
+
+        trace_json = trace.to_json()
+        matches = []
+        total_found = 0
+        for match in regex.finditer(trace_json):
+            if total_found >= max_matches:
+                break
+            matches.append(self._create_regex_match(match, trace_json))
+            total_found += 1
+
+        return SearchTraceRegexResult(
+            pattern=pattern,
+            total_matches=total_found,
+            matches=matches,
+        )
+
+    def _create_regex_match(
+        self, match: re.Match[str], text: str, span_id: str = "trace"
+    ) -> RegexMatch:
+        """Create a RegexMatch with surrounding context from a regex match object."""
+        matched_text = match.group()
+        start, end = match.span()
+        context_start = max(0, start - 100)
```

Comment:
Do you think that we should make this configurable or has your testing confirmed 100 characters to be an effective value?